app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarReading.kt

```diff
@@ -44,6 +44,17 @@ data class BloodSugarReading(val value: String, val type: BloodSugarMeasurementT
       }
     }
 
+  val displayValueMmol: String
+    get() {
+      return when (type) {
+        Random -> value.toFloat().toString()
+        PostPrandial -> value.toFloat().toString()
+        Fasting -> value.toFloat().toString()
+        HbA1c -> value.toFloat().toString()
+        is Unknown -> value.toInt().toString()
+      }
+    }
+
```

Comment:
We could change `displayValue` and `displayUnit` to functions which accept the `BloodSugarUnitPreference` as a parameter and then do the conversions within that. This will allow us to keep all the conditional logic in a single place instead of spreading it across multiple places.

Reply:
But didn't we talk about `BloodSugarReading` not knowing about `BloodSugarUnitPreference`? That's why I went with this approach. Changing `displayValue` and `displayUnit` to functions.

Reply:
No, what I meant was that the properties of `BloodSugarReading` should always be in `mg/DL`.

Reply:
Ah okay. Doing this change 👍🏽

Reply:
In this scenario, when we receive a `BloodSugarReading`, we have absolutely no idea if it's in mmol/L or `mg/DL`. Having functions that accept it as parameters is perfectly fine since those functions can be moved to other places later if needed.